### PR TITLE
Document using SaaS with internal registry

### PR DESCRIPTION
**Pull request by xphyr**
Documentation on how to use the SaaS version of the Assisted Installer with an internal mirror registry. As called out in the documentation, it is not possible to be 100% reliant on the internal mirror, but the end resulting cluster will use an internal mirror to pull all images. This documentation is based on a request from @romfreiman 

Signed-off-by: Mark D <xphyr@users.noreply.github.com>

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [X] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] None

## How was this code tested?

- [x] No tests needed - documentation was tested by walking through the process multiple times in internal lab.

## Checklist

- [X] Title and description added to both, commit and PR.
- [X] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/user-guide/cloud-with-mirror.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+> **NOTE**: You must ensure that you have mirrored a copy of OCP that _EXACTLY_ matches the version that you are installing from the Assisted Installer service hosted at [console.redhat.com](https://console.redhat.com). eg. If you are installing OpenShift 4.11.5 from the Assisted Installer, you must have previously mirrored verion 4.11.5 to your local registry.
+
+## Environment Setup
+
+### User Authentication
+
+User Authentication is using JWT tokens. The JWT token is **valid for 15 minutes** and needs to be provided in the header of the HTTP request.
+
+First you must obtain your offline token from https://console.redhat.com/openshift/token (This token does not expire, so do not share it with others) and set it as OFFLINE_TOKEN, then generate your JWT_TOKEN:
```
Comment (omertuc) [inline docs/user-guide/cloud-with-mirror.md:41]: This is repeating documentation we already have in cloud.md

Comment (xphyr) [reply docs/user-guide/cloud-with-mirror.md]: @omertuc thanks for the feedback, I have updated this section and removed the duplicate documentation. Please let me know if you have any additional questions about this.

**Comment by adetalhouet** [comment]
is this duplicates this doc: https://github.com/openshift/assisted-service/blob/master/docs/user-guide/install-customization.md#discovery-ignition

**Pull request merged**